mirai-core-api/src/commonMain/kotlin/auth/BotAuthorization.kt

```diff
@@ -101,8 +104,115 @@ public interface BotAuthInfo {
     public val id: Long
     public val deviceInfo: DeviceInfo
     public val configuration: BotConfiguration
+
+    /**
+     * 是否是首次登录
+     *
+     * 首次登录指的是首次调用 [Bot.login] 进行登录，直到登录成功的过程。
+     *
+     * 若在首次登录过程中多次进入[认证流程][BotAuthorization.authorize]，则这流程些均被视为首次登录。
+     *
+     * @see BotAuthorization.authorize
+     */
     public val isFirstLogin: Boolean
-    public val reLoginCause: Throwable?
+
+    /**
+     * 导致重新进入认证流程的原因
+     *
+     * 重新进入认证流程一般是第二次或多次进入认证流程，除了 [FastLoginError]。
+     *
+     * 第一次进入认证流程时该值为 `null`
+     *
+     */
+    public val reAuthCause: ReAuthCause?
+}
+
+/**
+ * 导致进行重新与服务器认证的原因
+ *
+ * @see ForceOffline
+ * @see MsfOffline
+ * @see FastLoginError
+ * @see NetworkError
+ * @see Unknown
+ */
+@NotStableForInheritance
+public sealed class ReAuthCause(
+    public open val bot: Bot,
+    public open val message: String?
+) {
+    /**
+     * Bot 被挤下线
+     *
+     * 当 Bot 在其他客户端使用相同协议登录此账号时，会导致 Bot 掉线。
+     *
+     * 若设置了 [BotConfiguration.autoReconnectOnForceOffline] 则 mirai 会尝试重新登录。
+     *
+     * 被挤下线后当前的 session 会失效，登录时会重新与服务器认证 Bot 账号信息。
```

Comment:
```suggestion
     * 被挤下线后当前的 session 会失效，因此需要重新登录。
```
另外这句话看起来应该跟在 147 行之后.

Reply:
对于 session 的解释似乎是多余的